conjure-java-core/src/main/java/com/palantir/conjure/java/types/AliasGenerator.java

```diff
@@ -190,13 +190,66 @@ private static boolean isAliasOfDouble(AliasDefinition typeDef) {
                 && typeDef.getAlias().accept(TypeVisitor.PRIMITIVE).equals(PrimitiveType.DOUBLE);
     }
 
-    private static MethodSpec createDefaultConstructor(TypeName aliasTypeName) {
-        return MethodSpec.constructorBuilder()
-                .addModifiers(Modifier.PRIVATE)
-                .addStatement(
-                        "this($T.empty())",
-                        aliasTypeName instanceof ParameterizedTypeName ? Optional.class : aliasTypeName)
-                .build();
+    private static final class DefaultConstructorVisitor implements Visitor<Optional<MethodSpec>> {
+        private final TypeName aliasTypeName;
+
+        DefaultConstructorVisitor(TypeName aliasTypeName) {
+            this.aliasTypeName = aliasTypeName;
+        }
+
+        @Override
+        public Optional<MethodSpec> visitPrimitive(PrimitiveType value) {
+            return Optional.empty();
+        }
+
+        @Override
+        public Optional<MethodSpec> visitOptional(OptionalType value) {
+            return Optional.of(MethodSpec.constructorBuilder()
+                    .addModifiers(Modifier.PRIVATE)
+                    .addStatement(
+                            "this($T.empty())",
+                            aliasTypeName instanceof ParameterizedTypeName ? Optional.class : aliasTypeName)
+                    .build());
+        }
+
+        @Override
+        public Optional<MethodSpec> visitList(ListType value) {
+            return Optional.of(MethodSpec.constructorBuilder()
+                    .addModifiers(Modifier.PRIVATE)
+                    .addStatement("this($T.emptyList())", Collections.class)
+                    .build());
+        }
+
+        @Override
+        public Optional<MethodSpec> visitSet(SetType value) {
+            return Optional.of(MethodSpec.constructorBuilder()
+                    .addModifiers(Modifier.PRIVATE)
+                    .addStatement("this($T.emptySet())", Collections.class)
+                    .build());
+        }
+
+        @Override
+        public Optional<MethodSpec> visitMap(MapType value) {
+            return Optional.of(MethodSpec.constructorBuilder()
+                    .addModifiers(Modifier.PRIVATE)
+                    .addStatement("this($T.emptyMap())", Collections.class)
+                    .build());
+        }
+
+        @Override
+        public Optional<MethodSpec> visitReference(com.palantir.conjure.spec.TypeName value) {
+            return Optional.empty();
+        }
```

Comment:
maybe it's too far-fetched to bother supporting, but in theory should we be recursing here? like if some defines an `Alias` of an `Alias` of a `List<String>`? (like conjure's DealiasingTypeVisitor)

Reply:
Ya, I didn't want to bother in this change since it's a much smaller subset, but it might make sense in a follow-up.

Reply:
maybe worth dropping a comment or a TODO in there so that people know we're aware this isn't the 100% solution?